spec/controllers/cloud_network_controller_spec.rb

```diff
@@ -135,7 +137,9 @@
       it "queues the create action" do
         expect(MiqTask).to receive(:generic_action_with_callback).with(task_options, queue_options)
         post :create, :params => { :button => "add", :format => :js, :name => 'test',
-                                   :tenant_id => 'id', :ems_id => @ems.id }
+          :cloud_tenant => {:id => 'id'}, :enabled => true, :external_facing => false, :shared => false,
```

Comment:
You've removed `ems_id` here, but it is still expected.

Also, since the `cloud_tenant` change, you need the `id` to be valid, so...

```diff
diff --git a/spec/controllers/cloud_network_controller_spec.rb b/spec/controllers/cloud_network_controller_spec.rb
index c7d057378..9c52b6bc6 100644
--- a/spec/controllers/cloud_network_controller_spec.rb
+++ b/spec/controllers/cloud_network_controller_spec.rb
@@ -124,10 +124,13 @@ describe CloudNetworkController do
           :role        => 'ems_operations',
           :zone        => @ems.my_zone,
           :args        => [{:name => "test", :admin_state_up => true, :shared => false,
-             :external_facing => false, :provider_network_type => 'vxlan', :tenant_id => 'íd',
+             :external_facing => false, :provider_network_type => 'vxlan', :tenant_id => cloud_tenant.ems_ref,
              :vlan_transparent => false}]
         }
       end
+      let(:cloud_tenant) do
+        FactoryGirl.create(:cloud_tenant)
+      end
 
       it "builds create screen" do
         post :button, :params => { :pressed => "cloud_network_new", :format => :js }
@@ -136,10 +139,20 @@ describe CloudNetworkController do
 
       it "queues the create action" do
         expect(MiqTask).to receive(:generic_action_with_callback).with(task_options, queue_options)
-        post :create, :params => { :button => "add", :format => :js, :name => 'test',
-          :cloud_tenant => {:id => 'id'}, :enabled => true, :external_facing => false, :shared => false,
-          :provider_network_type => 'vxlan', :id => 'new', :vlan_transparent => false,
-          :controller => "cloud_network"}
+        post :create, :params => {
+          :button                => "add",
+          :cloud_tenant          => {:id => cloud_tenant.id},
+          :controller            => "cloud_network",
+          :ems_id                => @ems.id,
+          :enabled               => true,
+          :external_facing       => false,
+          :format                => :js,
+          :id                    => 'new',
+          :name                  => 'test',
+          :provider_network_type => 'vxlan',
+          :shared                => false,
+          :vlan_transparent      => false,
+        }
       end
     end
   end
```

*almost* fixes the spec fail :).


What remains is this difference in the spec failure:

```diff
       -     :tenant_id=>"ems_ref_0000000000008",
       -     :vlan_transparent=>false}]}]
       +     :tenant_id=>"ems_ref_0000000000008"}]}]
```

which seems to be an actual bug... Not sure if you're supposed to be sending it or not, but.. it's only mentioned in specs and 1 place where you set it to false.

```
spec/javascripts/components/cloud_network/cloud_network_form_spec.js
54:        vlan_transparent: false

spec/controllers/cloud_network_controller_spec.rb
128:             :vlan_transparent => false}]
154:          :vlan_transparent      => false,

app/assets/javascripts/components/cloud_network/cloud-network-form.js
42:      vm.cloudNetworkModel.vlan_transparent = false;
```

Reply:
@himdel, thanks for putting me back on tracks, I was looking everywhere but in the right place.
Ignore the vlan_transparent, I added that one exhaustively.